### PR TITLE
[stable/fluentd] Fix missing serviceName in HPA

### LINKED ISSUE #21201: [stable/fluentd] Error when autoscaling is enabled
**Describe the bug**
When installing the chart with:

`helm install fluentd stable/fluentd  --set autoscaling.enabled=true`

I get the following error:

`error validating "": error validating data: ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec`

**Version of Helm and Kubernetes**:
```
$ helm version
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}
```
EKS 1.14

**Which chart**:
https://github.com/helm/charts/tree/master/stable/fluentd

**What happened**:
Can't install chart with HPA enabled.

**What you expected to happen**:
Can install chart with HPA enabled.

**How to reproduce it** (as minimally and precisely as possible):

`helm install fluentd stable/fluentd  --set autoscaling.enabled=true`

### PR DESCRIPTION
**Pull request by jmthvt**
#### Which issue this PR fixes

Fixes https://github.com/helm/charts/issues/21201

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/fluentd/templates/hpa.yaml [hunk 1]
```diff
@@ -7,12 +7,13 @@ metadata:
...
     chart: {{ template "fluentd.chart" . }}
     release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}    
+    heritage: {{ .Release.Service }}
 spec:
   scaleTargetRef:
     apiVersion: apps/v1
     kind: StatefulSet
     name: {{ include "fluentd.fullname" . }}
+    serviceName: {{ template "fluentd.serviceName" . }}
```
Comment (hectorj2f) [inline stable/fluentd/templates/hpa.yaml]: Why don't you do all the logic in one line here? You can have a default in case the value isn't set

Comment (jmthvt) [reply stable/fluentd/templates/hpa.yaml]: @hectorj2f That's amended, PTAL

**Comment by jmthvt** [comment]
bump

**Comment by zanhsieh** [comment]
@rendhalver
@miouge1
@hectorj2f
Would you guys LGTM this PR please? Thank you.

**Comment by eaterm** [comment]
We have had no issues with the chart until this got merged:
Now we see the following when the HPA is enabled

`unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(HorizontalPodAutoscaler.spec.scaleTargetRef): unknown field "serviceName" in io.k8s.api.autoscaling.v2beta2.CrossVersionObjectReference`

**Comment by Jonher937** [comment]
We are seeing this too.
Looking at the docs I can't see the field either: https://godoc.org/k8s.io/api/autoscaling/v2beta2

```
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(HorizontalPodAutoscaler.spec.scaleTargetRef): unknown field "serviceName" in io.k8s.api.autoscaling.v2beta2.CrossVersionObjectReference

helm.go:84: [debug] error validating "": error validating data: ValidationError(HorizontalPodAutoscaler.spec.scaleTargetRef): unknown field "serviceName" in io.k8s.api.autoscaling.v2beta2.CrossVersionObjectReference
```

**Comment by jmthvt** [comment]
Sorry all, this PR was wrong and what I was attempted to fix got merged in the meantime.
We need to revert.. :/

**Pull request merged**